dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp

```diff
@@ -261,6 +263,122 @@ bool DMFileReader::getSkippedRows(size_t & skip_rows)
     return next_pack_id < use_packs.size();
 }
 
+bool DMFileReader::skipNextBlock(size_t skip_rows [[maybe_unused]])
+{
+    // Go to next available pack.
+    size_t skip;
+    if (!getSkippedRows(skip))
+        return false;
+
+    // The next contiguous packs will be read in next read, mark them as not used.
+    size_t read_pack_limit = read_one_pack_every_time ? 1 : 0;
+    const std::vector<RSResult> & handle_res = pack_filter.getHandleRes();
+    RSResult expected_handle_res = handle_res[next_pack_id];
+    auto & use_packs = pack_filter.getUsePacks();
+    size_t start_pack_id = next_pack_id;
+    const auto & pack_stats = dmfile->getPackStats();
+    size_t read_rows = 0;
+    for (; next_pack_id < use_packs.size() && use_packs[next_pack_id] && read_rows < rows_threshold_per_read; ++next_pack_id)
+    {
+        if (read_pack_limit != 0 && next_pack_id - start_pack_id >= read_pack_limit)
+            break;
+        if (enable_handle_clean_read && handle_res[next_pack_id] != expected_handle_res)
+            break;
+
+        read_rows += pack_stats[next_pack_id].rows;
+        scan_context->total_dmfile_skipped_packs += 1;
+
+        // Mark this pack as not used.
```

Comment:
Why we need to mark this pack unused? Will `use_packs[next_pack_id]` still be used in future calculation？

Reply:
`use_packs` is reference of use_packs in `DMFileReader`. DMFileReader will use it to judge whether a pack should be  read.